modules/ROOT/pages/secure-storage.adoc

```diff
@@ -0,0 +1,110 @@
+= Secure Storage Overview
+
+Anypoint Studio 7.10 and earlier versions stored the information of authenticated users in the workspace, meaning that you could have different authenticated users per workspace.
+
+Studio 7.11 introduces a Secure Storage facility provided by Eclipse to encrypt and save your credentials.
+
+Studio provides a master password provider that automatically generates a master password for all your user credentials. When adding a new credential, Studio uses your master password to encrypt and save that credential into a secure storage file in your file system.
+
+Because of the architecture of the secure storage, the authenticated users are now shared across all Eclipse-based workspaces and Anypoint Studio instances.
+
+== Master Password Providers
+
+The available master password providers depend on your operative system. Studio assigns a priority to the available master password providers and enables you to choose which one to use. By default, Studio assigns a higher level of priority to the password provider that uses your OS's underlying capabilities to generate and encrypt your master password.
```

Comment:
Studio inherits this from Eclipse. The password does not change until the user chooses to re-generate it, and the password is never shown to the user. 
This is a behavior inherited from Eclipse and the user has no control over it, except when choosing to ignore the prioritization that "Studio" (Eclipse actually) applies to each password provider. -Which in my experience is changing from the inherited OS dependant provider to the UI where the user can define a password of his own.-

The term "password hint" seems broken as well. Those are actually security questions in case the master password is lost (which I think can rarely happen if the password is always self-generated from the OS login information, and even when clearing cache, the password self-generates right after doing so).

@mdaloia Am I misunderstanding something about this behavior?

Reply:
Hi, the behavior and config options as Fer described are correct.